app/src/main/java/org/xbmc/kore/ui/volumecontrollers/VolumeControllerDialogFragment.java

```diff
@@ -0,0 +1,172 @@
+    };
+    private long lastHardwareVolumeKeyEvent;
+    private Handler callbackHandler = new Handler();
+    private Runnable dismissDialog = new Runnable() {
+        @Override
+        public void run() {
+            long timeSinceLastEvent = System.currentTimeMillis() - lastHardwareVolumeKeyEvent;
+            if (timeSinceLastEvent >= AUTO_DISMISS_DELAY) {
+                Dialog dialog = getDialog();
+                if (dialog != null && dialog.isShowing()) {
+                    dialog.dismiss();
+                }
+            }
+        }
+    };
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        if (volumeKeyActionHandler == null) {
+            volumeKeyActionHandler = new VolumeKeyActionHandler(hostManager, getContext(), this);
+        }
+        getDialog().setOnKeyListener(new DialogInterface.OnKeyListener() {
+            @Override
+            public boolean onKey(android.content.DialogInterface dialog, int keyCode,
+                    android.view.KeyEvent event) {
+
+                return volumeKeyActionHandler.handleDispatchKeyEvent(event);
+            }
+        });
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+            @Nullable Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.volume_controller_dialog, container, false);
+        ButterKnife.inject(this, rootView);
+
+        return rootView;
+    }
+
+    @Override
+    public void show(FragmentManager manager, String tag) {
+        super.show(manager, tag);
+
+        lastHardwareVolumeKeyEvent = System.currentTimeMillis();
+        delayedDismissDialog();
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        hostManager = HostManager.getInstance(getContext());
+
+        setListeners();
+
+        registerObserver();
+    }
+
+    private void registerObserver() {
+        HostConnectionObserver hostConnectionObserver = hostManager.getHostConnectionObserver();
+        if (hostConnectionObserver == null) {
+            return;
+        }
+
+        hostConnectionObserver.registerApplicationObserver(this, true);
+        hostConnectionObserver.forceRefreshResults();
+    }
+
+    private void setListeners() {
+        volumeMuteButton.setOnClickListener(onMuteToggleOnClickListener);
+        volumeMutedIndicatorButton.setOnClickListener(onMuteToggleOnClickListener);
+
+        volumeLevelIndicator.setOnVolumeChangeListener(
+                new VolumeLevelIndicator.OnVolumeChangeListener() {
+                    @Override
+                    public void onVolumeChanged(int volume) {
+                        cancelDismissDialog();
+                        new Application.SetVolume(volume).execute(hostManager.getConnection(),
+                                defaultIntActionCallback, new Handler());
```

Comment:
A new Handler shouldn't be created here. `callBackHandler` should be used.
PS. I get that you got this code from somewhere else (`NowPlayingFragment`), but it is also wrong there. I'll fix it.